packages/vm/test/api/EIPs/eip-4200-static-relative.spec.ts

```diff
@@ -0,0 +1,222 @@
+    // Includes SSTOREs so we are sure we jump to the right location
+    for (let jumptableSize = 2; jumptableSize <= 255; jumptableSize++) {
+      let eCode = ''
+      const arr: number[] = []
+      for (let i = 0; i < jumptableSize; i++) {
+        arr.push(i * 6)
+      }
+      eCode += getRJUMPVCode(arr)
+      for (let i = 0; i < jumptableSize; i++) {
+        const index = i.toString(16).padStart(2, '0')
+        // PUSH index PUSH 0 SSTORE STOP
+        eCode += '60' + index + '60005500'
+      }
+      for (let jumptableIndex = 0; jumptableIndex < jumptableSize; ) {
+        const jumpCase = jumptableIndex.toString(16).padStart(2, '0')
+        const code = '60' + jumpCase + eCode
+        const { result } = await runTx(vm, getEOFCode(code), nonce++)
+        const value = await vm.stateManager.getContractStorage(
+          result.createdAddress!,
+          Buffer.from('00'.repeat(32), 'hex')
+        )
+        const expected = jumptableIndex === 0 ? '' : jumptableIndex.toString(16).padStart(2, '0')
+        st.ok(
+          value.toString('hex') === expected,
+          'rjumpv size ' + jumptableSize + ' index ' + jumptableIndex + ' valid'
+        )
+        jumptableIndex += Math.ceil(Math.random() * 30)
+      }
+    }
+  })
+
+  t.test('eip-4200 is invalid opcode in legacy bytecode', async (st) => {
+    const vm = await VM.create({ common })
+    const account = await vm.stateManager.getAccount(sender)
+    const balance = GWEI * BigInt(21000) * BigInt(10000000)
+    account.balance = balance
+    await vm.stateManager.putAccount(sender, account)
+    let nonce = 0
+
+    const codes = [
+      ['0x5C000000', 'RJUMP'],
+      ['0x60015D000000', 'RJUMPI'],
+      ['0x6000' + getRJUMPVCode([0]) + '00', 'RJUMPV'],
+    ]
+
+    for (const code of codes) {
+      const { result } = await runTx(vm, code[0], nonce++)
+      st.ok(result.execResult.exceptionError!.error === 'invalid opcode', code[1])
+    }
+  })
+  t.test('eip-4200 invalid eof format tests', async (st) => {
+    const vm = await VM.create({ common })
+    const account = await vm.stateManager.getAccount(sender)
+    const balance = GWEI * BigInt(21000) * BigInt(10000000)
+    account.balance = balance
+    await vm.stateManager.putAccount(sender, account)
+    let nonce = 0
+
+    const codes = [
+      ['5C00', 'truncated RJUMP'],
+      ['5D00', 'truncated RJUMPI'],
+      ['5E0100', 'truncated RJUMPV'],
+      ['5C' + getInt16Str(2) + '5B005C' + getInt16Str(-5), 'RJUMP as final instruction'],
+      ['5C' + getInt16Str(2) + '5B0060015D' + getInt16Str(-7), 'RJUMPI as final instruction'],
+      ['5C' + getInt16Str(2) + '5B0060005B' + getRJUMPVCode([-9]), 'RJUMPV as final instruction'],
+      ['5C' + getInt16Str(1) + '00', 'RJUMP targets code outside container'],
+      ['60015D' + getInt16Str(1) + '00', 'RJUMPI targets code outside container'],
+      ['6000' + getRJUMPVCode([1]) + '00', 'RJUMPV targets code outside container'],
```

Comment:
Can we add negative target-out-of-code cases here as well?

Reply:
Also a good point, not sure why I missed this because I think this was in the OG test specs :thinking: